lite2/client.go

```diff
@@ -89,6 +89,11 @@ func SkippingVerification(trustLevel tmmath.Fraction) Option {
 // current primary is unavailable.
 func Witnesses(providers []provider.Provider) Option {
 	return func(c *Client) {
+		for _, witness := range providers {
+			if witness.ChainID() != c.ChainID() {
+				panic("Witness chainID is not equal to the Lite Client chainID")
```

Comment:
I think we probably want this function to add an error as a return type, rather than panic, no?

Reply:
Yeah that was my initial thought but I noticed that another function that creates an option uses panic. It would also mean changing the function type `Option` to return possible errors for all other options. Let me know if you want me to make the change and I can do it.

Reply:
does it recover somewhere or causes the node to panic and halt?